src/Symfony/Bundle/FrameworkBundle/Secrets/SodiumVault.php

```diff
@@ -0,0 +1,164 @@
+    {
+        if (!$override && (file_exists($this->secretsDir.$this->encryptionKeyName) || file_exists($this->secretsDir.$this->decryptionKeyName))) {
+            return false;
+        }
+
+        $this->decryptionKey = sodium_crypto_box_keypair();
+        $this->encryptionKey = sodium_crypto_box_publickey($this->decryptionKey);
+
+        $this->export($this->encryptionKeyName, $this->encryptionKey);
+        $this->export($this->decryptionKeyName, $this->decryptionKey);
+
+        return true;
+    }
+
+    public function seal(string $name, string $secret): void
+    {
+        $this->loadKeys($name);
+        $this->export($name.'.sodium', sodium_crypto_box_seal($secret, $this->encryptionKey));
+    }
+
+    public function reveal(string $name): ?string
+    {
+        $this->loadKeys($name);
+
+        if (!file_exists($file = $this->secretsDir.$name.'.sodium')) {
+            return null;
+        }
+
+        if (!\is_string($sealedSecret = include $file)) {
+            throw new \LogicException(sprintf('Cannot load secret "%s".', $name));
+        }
+
+        return sodium_crypto_box_seal_open($sealedSecret, $this->decryptionKey);
+    }
+
+    public function list(bool $reveal = false): array
+    {
+        $this->loadKeys();
+        $secrets = [];
+
+        foreach (scandir($this->secretsDir) as $name) {
+            if (0 === substr_compare($name, '.sodium', -\strlen('.sodium'))) {
+                $name = substr($name, 0, -\strlen('.sodium'));
+                $secrets[$name] = $reveal ? $this->reveal($name) : '********';
+            }
+        }
+
+        return $secrets;
+    }
+
+    public function zero(string &...$sensitiveData): void
+    {
+        foreach ($sensitiveData as &$data) {
+            if (\extension_loaded('sodium')) {
+                sodium_memzero($data);
+            } elseif (\extension_loaded('libsodium')) {
+                \Sodium\memzero($data);
+            } else {
+                for ($i = 0; $i < \strlen($data); ++$i) {
+                    $data[$i] = "\0";
+                }
+                $data = null;
+            }
+        }
+    }
+
+    public function __destruct()
+    {
+        if (null !== $this->decryptionKey) {
+            $this->zero($this->encryptionKey, $this->decryptionKey);
+        }
+    }
+
+    private function loadKeys(string $name = null): void
+    {
+        if (null !== $name && preg_match('/[^-._A-Za-z0-9\x80-\xFF]/', $name)) {
+            throw new \LogicException('Secret name is invalid: "%s".', $name);
+        }
+
+        if (null !== $this->decryptionKey) {
+            return;
+        }
+
+        if (!file_exists($this->secretsDir.$this->encryptionKeyName)) {
+            throw new \LogicException('Encryption key not found in "%s".', $this->secretsDir);
+        }
+
+        if (!file_exists($this->secretsDir.$this->decryptionKeyName)) {
+            throw new \LogicException('Decryption key not found in "%s".', $this->secretsDir);
```

Comment:
private key is not required to encrypt secret (that should also be the default case IMHO)
- developers have public key only
- server have private key only
- ops/leader have both in order to rotate keys

Reply:
renamed back to sodium.encrypt/decryt.key